tests/semver/test_constraint_utils.py

```diff
@@ -0,0 +1,62 @@
+import pytest
+
+import poetry.semver
+from poetry.semver.constraint_utils import is_constraint
+from poetry.semver.constraint_utils import sorted_constraints
+
+# TODO: test PEP-0440 vs. SEM-VER constraints
+
+
+# from test_main.py
+@pytest.mark.parametrize(
+    "constraint,result",
+    [
+        ("*", True),
+        ("*.*", True),
+        ("v*.*", True),
+        ("*.x.*", True),
+        ("x.X.x.*", True),
+        ("!=1.0.0", True),
+        (">1.0.0", True),
+        ("<1.2.3", True),
+        ("<=1.2.3", True),
+        (">=1.2.3", True),
+        ("=1.2.3", True),
+        ("1.2.3", True),
+        ("=1.0", True),
+        ("1.2.3b5", True),
+        (">= 1.2.3", True),
+        (">dev", True),
+        ("n.a.n", False),
```

Comment:
same here.

Reply:
same comment as above, these are already in poetry tests; only added an `n.a.n` to be sure that it doesn't match a constraint because it's all chars but otherwise matches the pattern.